docs/src/main/asciidoc/kubernetes-resources.adoc

```diff
@@ -139,3 +139,36 @@ By default the Kubernetes resources do not contain readiness and liveness probes
 
 The values of the generated probes will be determined by the configured health properties: `quarkus.smallrye-health.root-path`, `quarkus.smallrye-health.liveness-path` and `quarkus.smallrye-health.readiness-path`.
 More information about the health extension can be found in the relevant link:health-guide[guide].
+
+== Tuning the generated resources using application.properties
+
+The kubernetes extension allows tuning the generated manifest, using the `application.properties` file.
+Here are some examples:
+
+=== Changing the number of replicas:
+To change the number of replicas from 1 to  3:
+
+[source]
+----
+quarkus.kubernetes.replicas=3 
+----
+
+=== Adding labels:
+To add a new label to all generated resources, say `foo=bar`:
+
+[source]
+----
+quarkus.kubernetes.labels[0].key=foo
+quarkus.kubernetes.labels[0].value=bar
```

Comment:
I wonder if it would possible to leverage our config map support instead of doing this?

Reply:
Config map support?

Reply:
Well, we have two sorts of configuration in the extensions:
- configuration that is properly mapped by Quarkus - I understood after reading the end of the diff that this is not the case for this one
- passthrough configuration, which looks more like this case.

In the case of passthrough configuration, we usually don't use the `quarkus` prefix because it's not Quarkus config and it's not consumed by Quarkus (note that there's a good chance you will have warnings if your config is not properly consumed by Quarkus).

Typically, we have an example of that in the Kafka Streams extension. They have `quakus.kakfa-streams.` properties for everything Quarkus and `kafka-streams.` properties for everything passthrough.

Even in the passthrough case, I think we should follow the Quarkus case convention.

Reply:
So this one is a little bit of a hybrid:

It does have some of its own configuration (see `io.quarkus.kubernetes.deployment.KubernetesConfig`) but it also has some passthrough properties.
From a user's point of view this split doesn't matter and shouldn't be exposed it it IMHO.

I fully agree with the case convention - it's best to keep thing consistent.

Reply:
Well, I took the example of Kafka Streams for a reason: it has both and both have a different namespace because they are not the same thing. It's exactly the same case we have here.

And yes, the user should know the difference because the passthrough properties won't be documented the same way (if they ever are).

Hiding the fact that they are different is a bad idea because, in the end, it makes a difference for the user.

Reply:
I still think that usability trumps documentation so I don't agree with the assessment. 
However since we already do it the other way in Kafka, I think we should do it here as well.
@iocanel would you like to use something like `kubernetes` as the root of these "passthrough" properties? Also it would be great if the Quarkus documentation pointed to the relevant Dekorate documentation where these properties are mentioned.

Reply:
It's not more usable if the user doesn't understand what they do.

These properties are inherently different.

Reply:
So, should I keep the original prefix `dekorate` or completely wipe it?

Reply:
We definitely need to drop `quarkus` for these passthrough properties. 
I think for users it makes sense then to have: 

```
kubernetes.labels[0].key=foo
```

Reply:
Removed both quarkus and dekorate prefixes.